src/complex.jl

```diff
@@ -0,0 +1,285 @@
+imaginary part, return the original variable as it was defined by the user.
+
+See also [`conj`](@ref conj), [`real`](@ref), [`imag`](@ref).
+"""
+ordinary_variable(x::AbstractVariable) = x
+
+"""
+    conj(x::AbstractVariable)
+
+Return the complex conjugate of a given variable if it was declared as a complex variable; else return the
+variable unchanged.
+
+    conj(x::AbstractMonomial)
+    conj(x::AbstractVector{<:AbstractMonomial})
+    conj(x::AbstractTerm)
+    conj(x::AbstractPolynomial)
+
+Return the complex conjugate of `x` by applying conjugation to all coefficients and variables.
+
+See also [`iscomplex`](@ref iscomplex), [`isconj`](@ref isconj).
+"""
+Base.conj(x::AbstractVariable) = x
+
+"""
+    real(x::AbstractVariable)
+
+Return the real part of a given variable if it was declared as a complex variable; else return the variable
+unchanged.
+
+    real(x::AbstractMonomial)
+    real(x::AbstractVector{<:AbstractMonomial})
+    real(x::AbstractTerm)
+    real(x::AbstractPolynomial)
+
+Return the real part of `x` by applying `real` to all coefficients and variables; for this purpose, every complex-valued
+variable is decomposed into its real- and imaginary parts.
+
+See also [`iscomplex`](@ref iscomplex), [`isrealpart`](@ref isrealpart), [`imag`](@ref).
+"""
+Base.real(x::AbstractVariable) = x
+
+"""
+    imag(x::AbstractVariable)
+
+Return the imaginary part of a given variable if it was declared as a complex variable; else return zero.
+
+    imag(x::AbstractMonomial)
+    imag(x::AbstractVector{<:AbstractMonomial})
+    imag(x::AbstractTerm)
+    imag(x::AbstractPolynomial)
+
+Return the imaginary part of `x` by applying `imag` to all coefficients and variables; for this purpose, every complex-valued
+variable is decomposed into its real- and imaginary parts.
+
+See also [`iscomplex`](@ref iscomplex), [`isimagpart`](@ref isimagpart), [`real`](@ref).
+"""
+Base.imag(::AbstractVariable) = MA.Zero()
+
+# extend to higher-level elements. We make all those type-stable (but we need convert, as the construction method may
+# deliver simpler types than the inputs if they were deliberately casted, e.g., term to monomial)
+function iscomplex(p::AbstractPolynomialLike)
+    for v in variables(p)
+        if !iszero(maxdegree(p, v)) && iscomplex(v)
+            return true
+        end
+    end
+    return false
+end
+Base.conj(x::M) where {M<:AbstractMonomial} =
+    convert(M, reduce(*, conj(var)^exp for (var, exp) in zip(variables(x), exponents(x)); init=constantmonomial(x)))
+Base.conj(x::V) where {V<:AbstractVector{<:AbstractMonomial}} = monovec(conj.(x))
+Base.conj(x::T) where {T<:AbstractTerm} = convert(T, conj(coefficient(x)) * conj(monomial(x)))
+Base.conj(x::P) where {P<:AbstractPolynomial} = nterms(x) == 0 ? x : convert(P, sum(conj(t) for t in x))
+
+real_coefficient_type_polynomial_like(::AbstractPolynomialLike{R}) where {R<:Real} = R
```

Comment:
```suggestion
real_coefficient_type(::AbstractPolynomialLike{R}) where {R<:Real} = R
```

Reply:
Why not just `real(coefficient_type(...))` ?

Reply:
True, because I was not aware that `real` also works on the types directly. But this is much cleaner, so I dropped the ugly function.